consensus/src/simplex/signing_scheme/ed25519.rs

```diff
@@ -147,11 +208,20 @@ impl Read for Certificate {
     }
 }
 
-impl signing_scheme::Scheme for Scheme {
-    type Signature = Ed25519Signature;
+impl<P: PublicKey> signing_scheme::Scheme for Scheme<P> {
+    type PublicKey = P;
```

Comment:
I find specifying a `PublicKey` for "peers" and a `Signature` for "votes" to be a bit confusing (although it isn't "wrong" per se).